scripts/generate_dump

```diff
@@ -26,15 +35,96 @@ DO_COMPRESS=true
 CMD_PREFIX=
 SINCE_DATE="@0" # default is set to January 1, 1970 at 00:00:00 GMT
 REFERENCE_FILE=/tmp/reference
+TECHSUPPORT_TIME_INFO=/tmp/techsupport_time_info
 BASE=sonic_dump_`hostname`_`date +%Y%m%d_%H%M%S`
 DUMPDIR=/var/dump
 TARDIR=$DUMPDIR/$BASE
 TARFILE=$DUMPDIR/$BASE.tar
 LOGDIR=$DUMPDIR/$BASE/dump
 NUM_ASICS=1
+HOME=${HOME:-/root}
+USER=${USER:-root}
+TIMEOUT_MIN="5"
+SKIP_BCMCMD=0
+
+save_bcmcmd() {
+    local start_t=$(date +%s%3N)
+    local end_t=0
+    local cmd="$1"
+    local filename=$2
+    local filepath="${LOGDIR}/$filename"
+    local do_gzip=${3:-false}
+    local tarpath="${BASE}/dump/$filename"
+    local timeout_cmd="timeout --foreground ${TIMEOUT_MIN}m"
```

Comment:
>timeout [](start = 23, length = 7)

`bcmcmd` support timeout `-t`. Could you use that instead of `timeout` command?

Reply:
@qiluo-msft 
We can't predict the behaviour of 'bcmcmd' in case syncd is crashed/stuck/in bad shape. The change you are suggesting might require very extensive testing.

Reply:
This is the my only concern, others lgtm.

---
In reply to: [546269345](https://github.com/Azure/sonic-utilities/pull/1249#discussion_r546269345) [](ancestors = 546269345)

Reply:
'timeout' supersedes the '-t' option of 'bcmcmd'. Is the concern regarding 'bcmcmd' may end up in a bad state like leaving the socket open?

Reply:
`bcmcmd` is not well tested with signal termination. I am not sure.

---
In reply to: [550679331](https://github.com/Azure/sonic-utilities/pull/1249#discussion_r550679331) [](ancestors = 550679331,546269345)

Reply:
@qiluo-msft 
"Could you use that and maybe together with timeout command" --- The bcmcmd has a default timeout of 30 sec. If no argument is given using -t option, '-t 30' is automatically taken. So the outside 'timeout' command **is** being used together with bcmcmd timeout option.
```
root@sonic:/home/admin# bcmcmd
USAGE: bcmcmd [-f <sun_path>] -v <cmd>
  -v                         verbose mode
  -f                         domain socket filename, default /var/run/sswsyncd/sswsyncd.socket
  -t                         timeout in seconds, default 30
RETURN VALUE:
    0                        success
    5                        socket io error
   22                        invalid argument
   62                        timeout

root@sonic:/home/admin#
```